### ISSUE TITLE
UTF8MB4 sulu build not working with on InnoDB default config

### ISSUE THREAD

**Issue by alexander-schranz** [sulu/sulu#3293]
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | 1.5

### Actual Behavior

A key have a maximum of 767 bytes in innodb mysql the following columns are too large. The maximum would be 191.

Table                            | Column           | Length       
--------------------|-------------|----------
Category.orm.xml         | category_key   | 255           
Keyword.orm.xml          | keyword          | 255           
BaseCollection.orm.xml | collection_key | 255           
CollectionMeta.orm.xml | title                | 255           
CollectionType.orm.xml | collection_type_key | 255  
FileVersion.orm.xml       | mimeType       | 255          
FileVersion.orm.xml       | name              | 255         
FileVersionMeta.orm.xml | title              | 255          
FormatOptions.orm.xml | formatKey       | 255
BaseRoute.orm.xml        | path                | 255
BaseRoute.orm.xml        | entityClass      | 255
BaseRoute.orm.xml        | entityId           | 255
AccessControl.orm.xml  | entityClass      | 255
BaseUser.orm.xml          | email              | 255
Permission.orm.xml       | context           | 255
RoleSetting.orm.xml      | settingKey       | 255
Tag.orm.xml                  | name               | 255
Analytics.orm.xml          | webspace_key  | 255
Domain.orm.xml           | url                    | 255

```
SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 767 bytes
```

### Expected Behavior

Should also work with UTF8MB4

### Steps to Reproduce

Use the following config and run `bin/console sulu:build dev --destroy`

```yml
doctrine:
    dbal:
        default_connection: default
        connections:
            default:
                driver:   "%database_driver%"
                host:     "%database_host%"
                port:     "%database_port%"
                dbname:   "%database_name%"
                user:     "%database_user%"
                password: "%database_password%"
                charset:  utf8mb4
                default_table_options:
                    charset: utf8mb4
                    collate: utf8mb4_unicode_ci
                server_version: "%database_version%"
```

**Comment by alexander-schranz** [comment]
Think the only solution for this problem that we recommend here to set `innodb_large_prefix` option: https://dev.mysql.com/doc/refman/5.6/en/innodb-parameters.html#sysvar_innodb_large_prefix /cc @chirimoya

**Comment by danrot** [comment]
fixed in #3724
